src/scope.ml

```diff
@@ -257,7 +337,7 @@ let scope : mode -> sig_state -> env -> p_term -> tbox = fun md ss env t ->
         _Appl (_Appl s (scope env l)) (scope env r)
     | (P_Wrap(t)      , _         ) -> scope env t
   in
-  scope env t
+  scope env (add_implicits t)
```

Comment:
This cannot work, you only add implicit arguments at the toplevel of terms. This should be done recursively. The previous version actually did that.

Reply:
Now, add_implicits adds implicits not only at the toplevel, but also at the underneath layers, so it will work to do it like this.